### ISSUE TITLE
Turning on "Encrypt Index" option in Export window produces binary that can't be launched

### ISSUE THREAD

**Issue by enxas** [godotengine/godot#80804]
### Godot version

v4.1.1.stable.mono.official [bd6af8e0e]

### System information

Godot v4.1.1.stable.mono - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1660 Ti with Max-Q Design (NVIDIA; 31.0.15.3640) - Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz (12 Threads)

### Issue description

When "Encrypt Index" option is enabled in Export window exported binary throws error when launched. Turning "Encrypt Index" off will export properly and exported binary works as expected.
```
Error: Couldn't load project data at path ".". Is the .pck file missing?
If you've renamed the executable, the associated .pck file should also be renamed to match the executable's name (without the extension).
```
![image](https://github.com/godotengine/godot/assets/8630984/5c0e391c-b29b-4087-812e-6992e8f33595)

When binary is launched with ``--verbose`` flag, this is full error.

```
PS C:\Tmp\exported> .\expo.exe --verbose
PS C:\Tmp\exported> ERROR: The MD5 sum of the decrypted file does not match the expected value. It could be that the file is corrupt, or that the provided decryption key is invalid.
   at: (core\io\file_access_encrypted.cpp:94)
ERROR: Can't open encrypted pack directory.
   at: (core\io\file_access_pack.cpp:231)
ERROR: The MD5 sum of the decrypted file does not match the expected value. It could be that the file is corrupt, or that the provided decryption key is invalid.
   at: (core\io\file_access_encrypted.cpp:94)
ERROR: Can't open encrypted pack directory.
   at: (core\io\file_access_pack.cpp:231)
Error: Couldn't load project data at path ".". Is the .pck file missing?
If you've renamed the executable, the associated .pck file should also be renamed to match the executable's name (without the extension).
```

### Steps to reproduce

I have this configured:
Editor Settings -> Export -> Window -> C:/tools/rcedit.exe (from https://github.com/electron/rcedit/releases)

1. Generate key and add it to the environment variable like described in [Compiling with PCK encryption key](https://docs.godotengine.org/en/4.1/contributing/development/compiling/compiling_with_script_encryption_key.html).
2. Follow these first 3 commands (Build editor binary, Build export templates) in [Compiling with .NET](https://docs.godotengine.org/en/stable/contributing/development/compiling/compiling_with_dotnet.html#example-windows) to compile export templates with encryption key. (Use Godot 4.1.1 source code)
3. Create new project with MeshInstance3D and Camera that you gonna export.
4. Click Project -> Export. Add Windows Desktop.
5. In Custom Template section provide path to compiled templates (ex. C:/Godot/Godot-4.1.1/bin/godot.windows.template_debug.x86_64.mono.exe)
6. In Encryption tab turn on "Encrypt Exported PCK", "Encrypt Index" and provide "Encryption Key"
7. Click Export project and try to launch it with `--verbose` flag.
8. Export again with "Encrypt Index" turned off to make binary launch properly.

My Export settings (mostly default)
![261833767-dfc4b81b-49ac-4216-a0dd-b73f7c750661](https://github.com/godotengine/godot/assets/8630984/d9729cc0-b352-43d2-b71c-08c9bf42be99)

Remove .txt extension, couldn't upload this file othervise:
[export_presets.cfg.txt](https://github.com/godotengine/godot/files/12387360/export_presets.cfg.txt)

### Minimal reproduction project

NA

**Comment by Calinou** [comment]
Does it work if you don't set up rcedit, so that it doesn't modify the executable?

**Comment by enxas** [comment]
Yes, removing rcedit path from Editor Settings -> Export -> Windows still produces this error. Also unchecking "Modify Resources" doesn't change anything and still produces bad binary.

**Comment by bruvzg** [comment]
Usually this error happens when a wrong template is used. Re-check if you have set encryption key env. variable during template build (also check `core/script_encryption_key.gen.cpp` generated file after the build for the correct key).

**Comment by enxas** [comment]
File at `core/script_encryption_key.gen.cpp` contains this:
`uint8_t script_encryption_key[32]={0,0,0,0,0,0,0,0, 0,0,0,0,0,0,0,0, 0,0,0,0,0,0,0,0, 0,0,0,0,0,0,0,0};`
I assume these zeroes mean that encryption key env. variable wasn't properly set?

**Comment by bruvzg** [comment]
> I assume these zeroes mean that encryption key env. variable wasn't properly set?

Yes, zeros mean it's not set, it should be the key in hex format which is read from the `SCRIPT_AES256_ENCRYPTION_KEY` environment variable.

**Comment by enxas** [comment]
This is my Windows Terminal Powershell history. I did set the env. variable. Not sure why it didn't work.

![image](https://github.com/godotengine/godot/assets/8630984/f04ed98f-991b-4f00-a58a-c02a0178f36a)

Is there scons clean command to clean build artifacts or should I just delete `bin` directory and start over gain?

**Comment by bruvzg** [comment]
> Another question. If encryption key wasn't set properly, then why only "Encrypt Index" option is affected? "Encrypt Exported PCK" option works fine when enabled.

Have you selected any files to encrypt ("filters to include" in the export options encryption tab)? Only these files are encrypted, and an error will pop up only when one of the encrypted files is accessed.

**Comment by enxas** [comment]
> Have you selected any files to encrypt ("filters to include" in the export options encryption tab)?

No I haven't. I didn't knew that only explicitly listed files in "filters to include" are encrypted,

**Comment by ivanfemia** [comment]
I just experienced the same problem "Error: Couldn't load project data at path".". Is the .pck file missing? If you've renamed the executable, the associated.pck file should also be renamed to match the executable's name (without the extension)."

Tests conducted with a new blank project with one simple scene.

**Godot 4.2.1**

1. Exported with encryption, filter include (*.gd,*.tscn,*.env) and no **Encrypt index** -> WORKS
2. Added **Encrypt index** ->  ERROR

**Godot 4.3**

1. Exported with encryption and no **Encrypt index** -> WORKS
2. Added **Encrypt index** ->  ERROR

Any workaround?
What does the **Encrypt index** option do?

**Comment by DrJamgo** [comment]
Tested on Godot 4.3-stable, providing same filter and same key every time:
- Using **custom** template, **yes** file encryption, **yes** index encryption -> Same error as others
- Using **custom** template, **yes** file encryption, **no** index encryption -> WORKS
- Using **default** template, **yes** file encryption, **no** index encryption -> **WORKS as well?**

This really makes me wonder if the encryption **works at all** when not using the "Encrypt Index" option..

**Comment by bruvzg** [comment]
If it's not working, you have built the custom template incorrectly (without a key or with a wrong key). Keep in mind that only files specified by the filters on "Encryption" tab are encrypted. If you are using export template without encryption key (default, or incorrectly built), it might start, but will fail to load encrypted files, if index is encrypted it won't be able to find any files.

**Comment by DrJamgo** [comment]
I think you missed the point.. Other commenters already discovered that encrypting the index seem not to work.
I am suprised why it WORKS when I
- Use the default template
- Enable file encryption
- Set filter= `*.tscn,*.tres,*.gd`
- Set the Encryption Key: `e38f....`

And the game starts and works perfectly fine.. shouldnt it fail if it is encrypted?

**Comment by DrJamgo** [comment]
My point is that I can still run the exported game normally, when using the default template.

- Enable _File Encryption_ 
- Set the filter to `*.tscn,*.tres,*.gd`
- Set my Encryption Key `e38f...`

The game starts and works normally. I would expect it to fail instead.. 
Or did I somehow manage to make a game without using a single scene, resource or script?

**Comment by bruvzg** [comment]
4.3 apply filters as is (this is fixed in 4.4, it should automatically use apply encryption to remaps as well), so `*.tscn,*.tres,*.gd` will not match any files (these are source files and not included into the exported project):
 
- `.gd` are compiled to `.gds`
- `.tscn` are converted to binary `.scn`
- `.tres` are converted to binary `.res`
- all imported assets are converted to different internal formats.

**Comment by DrJamgo** [comment]
Ah okay, makes perfect sense.. thanks.

**Comment by SchimmelSpreu83** [comment]
I just figured out something strange, when I export my game with file encryption and encrypt index inside at the root directiory of my game (res://) it runs fine. But as soon as I place the exported game somewhere else, the game crashes with the same error.

**Comment by akien-mga** [comment]
The reports in this issue should all be due to using an export template that wasn't properly compiled with the encryption key. This may due to a user error, selecting the wrong binary, or somehow the build environment not having picked up the encryption key at build time.

I'm adding an info message in #107309 so that you can confirm in build logs whether the encryption key was properly found in the environment and used for a given build.
